src/psyclone/gocean1p0.py

```diff
@@ -1360,7 +1356,10 @@ def raw_arg_list(self):
                 arguments.append(arg.name)
             elif arg.type == "field":
                 # Field objects are Fortran derived-types
-                arguments.append(arg.name + "%data")
+                api_config = Config.get().api_conf("gocean1.0")
+                data = api_config.field_properties["go_grid_data"][0]\
```

Comment:
OK, so in this case we do actually want field properties, it's just that it happens that both field and grid objects have the raw data contained in an array called `data`. We shouldn't be using grid-configuration information to access the field property though. Either we have a new configuration option or we leave this part as it is for the moment (and open yet another issue).

Reply:
I see your point, let's discuss at telco tonight.

Reply:
See #676, postponing as a separate issue.

Reply:
Fine. Please could you add a TODO here and reference that issue.